frontend/src/components/TreeView.js

```diff
@@ -0,0 +1,207 @@
+import React, { useEffect, useState } from 'react'
+import { useNavigate, useLocation } from 'react-router-dom'
+import { Text, Box } from '@chakra-ui/react'
+const TreeView = (props) => {
+    const navigate = useNavigate()
+    const location = useLocation()
+    const { tickets } = props
+    const [tree, setTree] = useState([])
+    const pathname = location.pathname.substring(1).split('/')
+    const ticketType = pathname[0]
+    const ticketID = parseInt(pathname[1])
+
+    const sortAndTypeTickets = (list, type) => {
+        let newArrayWithType = list.map((obj) => ({ ...obj, type: type }))
+        let sortedArr = newArrayWithType.sort((a, b) =>
+            a.name > b.name ? 1 : -1
+        )
+        return sortedArr
+    }
+
+    let { SF, FI, PPR, UPR } = tickets
+
+    if (SF.length !== 0) {
+        SF = sortAndTypeTickets(SF, 'SF')
+        FI = sortAndTypeTickets(FI, 'FI')
+        PPR = sortAndTypeTickets(PPR, 'PPR')
+        UPR = sortAndTypeTickets(UPR, 'UPR')
+    }
+
+    useEffect(() => {
+        if (SF.length !== 0) {
+            let id = ticketID
+            if (ticketType === 'FI') {
```

Comment:
use a switch

also instead of hardcoding 'FI', 'PPR', etc. 
Change the const DATA_KEYS to be called TICKET_TYPES, and use that as an enum (so TICKET_TYPES.FI, etc.). this allows us to ensure the standardization of the abbreviations